x/wasm/internal/keeper/keeper.go

```diff
@@ -693,6 +702,58 @@ func (k Keeper) dispatchMessages(ctx sdk.Context, contractAddr sdk.AccAddress, i
 	return nil
 }
 
+// PinCode pins the wasm contract in wasmvm cache
+func (k Keeper) PinCode(ctx sdk.Context, codeID uint64) error {
+	codeInfo := k.GetCodeInfo(ctx, codeID)
+	if codeInfo == nil {
+		return sdkerrors.Wrap(types.ErrNotFound, "code info")
+	}
+
+	if err := k.wasmer.Pin(codeInfo.CodeHash); err != nil {
+		return sdkerrors.Wrap(types.ErrPinContractFailed, err.Error())
+	}
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.GetPinnedCodeIndexPrefix(codeID), []byte{})
```

Comment:
Oof, the handling of `[]byte{}` and `nil` inside `store.Set()` is a bit of a headache. Has *should* return true, but Get will return nil. 

Please just make this `[]byte{1}` to avoid potential issues later